### PR TITLE
107057: Tuning of queries related to export invitations to Excel.

### PR DESCRIPTION
**Pull request by trondlindanger**
[AB#107057](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/107057)

### CODE REVIEW

**Comment by janid1967** [comment]
Add number of PBI in front of title of PR
Add number of PBI in Description field in format AB#xxxxx

**Comment by trondlindanger** [comment]
> Add number of PBI in front of title of PR Add number of PBI in Description field in format AB#xxxxx

@janid1967 Done.

File: src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitationsQueryBase.cs [hunk 1]
```diff
@@ -50,58 +50,74 @@ public abstract class GetInvitationsQueryBase
 
             var ipoIdStartWith = GetIpoIdStartWith(filter.IpoIdStartsWith);
 
+            var filterPunchOutDates = new List<int>();
+            if (filter.PunchOutDates.Any())
+            {
+                filterPunchOutDates = filter.PunchOutDates.Select(x => (int)x).ToList<int>();
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitationsQueryBase.cs:56]: .ToList();

File: src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitationsQueryBase.cs [hunk 2]
```diff
@@ -50,58 +50,74 @@ public abstract class GetInvitationsQueryBase
...
+            }
+
+            var filterIpoStatuses = new List<int>();
+            if (filter.IpoStatuses.Any())
+            {
+                filterIpoStatuses = filter.IpoStatuses.Select(x => (int)x).ToList<int>();
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitationsQueryBase.cs:62]: .ToList();

File: src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitationsForExport/GetInvitationsForExcelQueryHandler.cs [hunk 3]
```diff
@@ -181,21 +181,27 @@ where personIds.Contains(p.Id)
...
                 if (project is null)
                 {
                     throw new ArgumentNullException(nameof(project));
                 }
 
-                var organizer = organizers.Single(x => x.Id == invitation.CreatedById);
+                var organizer = organizersDict[invitation.CreatedById];
                 var invitationWithIncludes = invitationsWithIncludes.Single(i => i.Id == invitation.Id);
-                var participants = invitationWithIncludes.Participants.ToList();
+                var participants = await Task.FromResult(invitationWithIncludes.Participants.ToList());
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitationsForExport/GetInvitationsForExcelQueryHandler.cs:204]: Why this change?

**Review by janid1967 — CHANGES_REQUESTED** [review]
Await feedback

File: src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitationsForExport/GetInvitationsForExcelQueryHandler.cs (hunk 3 above)
Comment (trondlindanger) [reply src/Equinor.ProCoSys.IPO.Query/GetInvitationsQueries/GetInvitationsForExport/GetInvitationsForExcelQueryHandler.cs:204]: Good question. No need for (the attempted) async enumeration here since the collection has already been loaded at this stage. Reverting back to previous implementation.

**Review by janid1967 — APPROVED** [review]
